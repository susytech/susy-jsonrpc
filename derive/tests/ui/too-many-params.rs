extern crate serde;
extern crate susy_jsonrpc_core;
#[macro_use]
extern crate susy_jsonrpc_derive;

use susy_jsonrpc_core::Result;

#[rpc]
pub trait Rpc {
	/// Has too many params
	#[rpc(name = "tooManyParams")]
	fn to_many_params(
        &self,
        a: u64, b: u64, c: u64, d: u64, e: u64, f: u64, g: u64, h: u64, i: u64, j: u64,
        k: u64, l: u64, m: u64, n: u64, o: u64, p: u64, q: u64,
    ) -> Result<String>;
}

fn main() {}