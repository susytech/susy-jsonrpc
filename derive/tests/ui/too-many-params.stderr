error: Maximum supported number of params is 16
  --> $DIR/too-many-params.rs:10:2
   |
10 |       /// Has too many params
   |  _____^
11 | |     #[rpc(name = "tooManyParams")]
12 | |     fn to_many_params(
13 | |         &self,
14 | |         a: u64, b: u64, c: u64, d: u64, e: u64, f: u64, g: u64, h: u64, i: u64, j: u64,
15 | |         k: u64, l: u64, m: u64, n: u64, o: u64, p: u64, q: u64,
16 | |     ) -> Result<String>;
   | |________________________^

error: aborting due to previous error

